spacy_methods.py

```diff
@@ -0,0 +1,213 @@
+        else:
+            count_dict[entity_sentence_check] = 1
+    return count_dict
+
+
+duplicate_items = entity_counter(entities)
+
+# Algorithm to assign new key values to 1, increment if already existing
+def append_to_array(key, value, dict_to_check):
+    if key in dict_to_check:
+        dict_to_check[key].append(value)
+    else:
+        dict_to_check[key] = [value]
+
+
+raw_entity_list = list(entities)
+
+
+def entity_indexer(lst):
+    ent_index_dict = {}
+
+    for i, entity_item in enumerate(lst):
+        entity_sentence_check = "{}".format(entity_item[0])
+
+        # entity_item[0] - Entity's name as a string
+        # entity_item[5] - the sentence number index
+
+        append_to_array(entity_item[0], entity_item[5], ent_index_dict)
+    return ent_index_dict
+
+
+ent_sentence_index = entity_indexer(raw_entity_list)
+
+
+def verb_matcher(txt):
+    # Verb Finder with Matcher
+    verb_matcher = Matcher(nlp.vocab)
+    verb_pattern = [{"POS": "VERB", "OP": "+"}]
+    verb_matcher.add("VERBS", [verb_pattern])
+    matches = verb_matcher(txt)
+
+    verb_information = []
+
+    for match in matches:
+
+        word_index = match[1]
+        original_verb = doc[match[1]]
+        lemmatized_verb = doc[match[1]].lemma_
+        verb_information.append([word_index, original_verb, lemmatized_verb])
+
+    return verb_information
+
+
+verbs = verb_matcher(doc)
+# print(verbs)
+
+# List of Verbs
+verbs = verb_matcher(doc)
+
+# List of Article's Sentences
+
+sentences = sentence_generator(doc)
+
+
+def verb_in_sentence(list_of_verbs, list_of_sentences):
+    # print(list_of_sentences)
+
+    nlp = spacy.load("en_core_web_md")
+    specific_verbs = []
+    sentence_count = -1
+    previous_sentence = None
+
+    word_count = 0
+
+    for i, sent in enumerate(list_of_sentences):
+        sentence_count += 1
+
+        # Check the sentence's text matches the previous sentence's
+        try:
+            doc = nlp(sent)
+            # print(doc)
+            sentence = doc.text
+
+            for token in doc:
+                word_count += 1
+                if token.pos_ == "VERB":
+
+                    # verb_index = word_count + token.i
+                    verb_text = token.text
+                    verb_lemma = token.lemma_
+                    sent_word_index = (
+                        token.i
+                    )  # Start index of the verb relative to the sentence
+
+                    # Document
+                    specific_verbs.append(
+                        (
+                            verb_text,
+                            verb_lemma,
+                            word_count,
+                            sentence,
+                            sentence_count,
+                            sent_word_index,
+                        )
+                    )
+        except:
+            continue
+
+    return specific_verbs
+
+
+the_verbs = verb_in_sentence(verbs, sentences)
+ppv = pprint.PrettyPrinter(indent=4)
```

Comment:
Same comment about moving it under `if __name__ == "__main__":`

Reply:
Moved. Resolved.